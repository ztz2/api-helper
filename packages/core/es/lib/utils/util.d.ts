import { JSONSchema4 } from 'json-schema';
import { APIHelper } from '../types';
export declare function checkType<T>(value: T, type: string): boolean;
export declare function isHttp<T>(value: T): boolean;
export declare function filterEmpty(value: any): any;
export declare function Try<T>(fn: () => T, err: (e: Error) => any): T;
export declare function arrayUniquePush(array: Array<any>, item: any): number;
export declare function uuid(): string;
export declare function randomId(): string;
export declare function mergeUrl(...args: string[]): string;
export declare function filterDesc(value?: string): string;
export declare function filterSchemaRoot(schemaList: Array<APIHelper.Schema>): ({
    params: never[];
    type: "string";
    rules: {
        required: boolean;
        minLength?: number | undefined;
        maxLength?: number | undefined;
        pattern?: string | undefined;
    };
    id: string;
    keyName: string;
    title: string;
    description: string;
    label: string;
    examples: string[];
    enum: (string | number)[];
} | {
    params: never[];
    type: "number";
    rules: {
        required: boolean;
        multipleOf?: number | undefined;
        minimum?: number | undefined;
        maximum?: number | undefined;
        exclusiveMinimum?: number | boolean | undefined;
        exclusiveMaximum?: number | boolean | undefined;
    };
    id: string;
    keyName: string;
    title: string;
    description: string;
    label: string;
    examples: string[];
    enum: (string | number)[];
} | {
    params: never[];
    type: "object";
    id: string;
    keyName: string;
    title: string;
    description: string;
    label: string;
    rules: {
        required: boolean;
    };
    examples: string[];
    enum: (string | number)[];
} | {
    params: never[];
    type: "array";
    rules: {
        required: boolean;
        minLength?: number | undefined;
        maxLength?: number | undefined;
        uniqueItems?: boolean | undefined;
    };
    id: string;
    keyName: string;
    title: string;
    description: string;
    label: string;
    examples: string[];
    enum: (string | number)[];
} | {
    params: never[];
    type: "boolean";
    id: string;
    keyName: string;
    title: string;
    description: string;
    label: string;
    rules: {
        required: boolean;
    };
    examples: string[];
    enum: (string | number)[];
} | {
    params: never[];
    type: "file";
    id: string;
    keyName: string;
    title: string;
    description: string;
    label: string;
    rules: {
        required: boolean;
    };
    examples: string[];
    enum: (string | number)[];
} | {
    params: never[];
    type: "null";
    id: string;
    keyName: string;
    title: string;
    description: string;
    label: string;
    rules: {
        required: boolean;
    };
    examples: string[];
    enum: (string | number)[];
} | {
    params: never[];
    type: "any";
    id: string;
    keyName: string;
    title: string;
    description: string;
    label: string;
    rules: {
        required: boolean;
    };
    examples: string[];
    enum: (string | number)[];
} | {
    params: never[];
    type: "unknown";
    id: string;
    keyName: string;
    title: string;
    description: string;
    label: string;
    rules: {
        required: boolean;
    };
    examples: string[];
    enum: (string | number)[];
})[];
export declare function filterSchemaRequired(schemaList: Array<APIHelper.Schema>): (APIHelper.IStringSchema | APIHelper.INumberSchema | APIHelper.IObjectSchema | APIHelper.IArraySchema | APIHelper.IBooleanSchema | APIHelper.IFileSchema | APIHelper.INullSchema | APIHelper.IAnySchema | APIHelper.IUnknownSchema)[];
export declare function filterKeyName(value?: string): string;
export declare function parserSchema(schema: JSONSchema4, parentSchema?: JSONSchema4, keyName?: string, memo?: Map<JSONSchema4, null>, options?: {
    autoGenerateId: boolean;
}): APIHelper.Schema | null;
export declare function processRequestSchema(requestDataSchema: APIHelper.Schema, requestSchemaRecord: Array<JSONSchema4>, requestJSONSchemaSource: JSONSchema4, keyNameMemo?: string[], options?: {
    autoGenerateId: boolean;
    callback?(parsedSchema: APIHelper.Schema): void;
}): APIHelper.IStringSchema | APIHelper.INumberSchema | APIHelper.IObjectSchema | APIHelper.IArraySchema | APIHelper.IBooleanSchema | APIHelper.IFileSchema | APIHelper.INullSchema | APIHelper.IAnySchema | APIHelper.IUnknownSchema | null;
export declare function uniqueRequestDataRootSchema(api: APIHelper.API): APIHelper.API;
export declare function deepAddSchemaRules(schema: null | APIHelper.Schema | APIHelper.SchemaList, rules?: Recordable): undefined;
export declare function processRequestSchemaPipeline(api: APIHelper.API, requestDataSchema: APIHelper.Schema, requestExtraDataSchema: APIHelper.Schema | null, options: Recordable): APIHelper.API;
export declare function processResponseSchemaPipeline(api: APIHelper.API, options: Recordable): void;
export declare function isSchemaObject(schema: APIHelper.Schema | null): boolean;
export declare function isSchemaPrimitiveValue(schema: APIHelper.Schema | null): boolean;
/**
 * @description 过滤原始值的Schema。保留纯粹的类型对象。原始值Schema用于TS类型申明有用，在生成JS对象，Class实体类时候，这些原始值类型则无用，需要过滤掉。
 * @example 例子说明：
    源数据：[
            { keyName: '', type: 'string' },
            { keyName: 'username', type: 'string' }
          ]
    过滤后：[
            { keyName: 'username', type: 'string' }
          ]
 * @param schema { schema: APIHelper.Schema | APIHelper.SchemaList | null } schema对象
 * @return APIHelper.Schema | APIHelper.SchemaList | null
 */
export declare function filterSchemaPrimitiveValue<T>(schema: APIHelper.Schema | APIHelper.SchemaList | null): T;
export declare function getErrorMessage<T extends string | Error & {
    msg?: string;
}>(error: T, prefix?: string, postfix?: string): string;
