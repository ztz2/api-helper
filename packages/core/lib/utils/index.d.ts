import * as util from './util';
import * as validator from './validator';
export { util, validator, };
