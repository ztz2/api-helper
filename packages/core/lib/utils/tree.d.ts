/**
 * 遍历 tree
 * @param {object[]} tree
 * @param {function} cb - 回调函数[(item: any) => unknown]
 * @param {string} children - 子节点 字段名
 * @param {string} mode - 遍历模式，DFS：深度优先遍历 BFS：广度优先遍历
 * @return {void}
 */
export declare function treeForEach(tree: any[] | undefined, cb: Function, children?: string, mode?: string): void;
/**
 * tree 转 数组
 * @param {object[]} tree
 * @param {string} options.children - 子节点 字段名
 * @param {string} options.mode - 遍历模式，DFS：深度优先遍历 BFS：广度优先遍历
 * @param {boolean} options.clear - 是否清楚子集
 * @return {array}
 */
export declare function treeToList(tree?: never[], options?: {}): any[];
/**
 * 数组 转 tree
 * @param {object[]} list
 * @param {object} options
 * @param {string|number|null|undefined} options.rootID - 根节点ID
 * @param {string|number|function} options.id - 唯一标识 字段名
 * @param {string|number|function} options.pid - 父节点ID 字段名
 * @param {string} options.children - 子节点 字段名
 * @return {array}
 */
export declare function listToTree(list?: never[], options?: {}): any;
/**
 * 树结构进行ID进行过滤，返回选中的ID数据结构
 * @param {object[]} tree
 * @param {Array<string>} 选中的节点ID
 * @param {object[]} tree 选中ID的树结构，包含所有父节点
 */
export declare function treeFilterByIds(tree?: never[], ids?: never[]): any;
/**
 * tree 寻找所有父级几点
 * @param {object[]} tree
 * @param {treeNode} node - 当前节点
 * @param {string|number|null|undefined} options.rootID - 根节点ID
 * @param {string|number|function} options.id - 唯一标识 字段名
 * @param {string|number|function} options.pid - 父节点ID 字段名
 * @param {string} options.children - 子节点 字段名
 * @return {array}
 */
export declare function treeFindParentNodes(tree: never[] | undefined, node: any, options: any): any;
export declare function treeMap(tree: any[] | undefined, cb: Function, children?: string, updateChildrenKey?: string): any[];
