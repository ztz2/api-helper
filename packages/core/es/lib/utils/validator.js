var rules = new Map();
rules.set('必须包含 paths 属性', function (document) {
    try {
        if ('paths' in document) {
            return true;
        }
        // 可能不是对象类型
    }
    catch (_a) {
        return false;
    }
});
export function validateOpenAPIDocument(document) {
    if (!document) {
        return false;
    }
    var errors = [];
    rules.forEach(function (rule, errorMeg) {
        var res = rule(document);
        if (!res) {
            errors.push(errorMeg);
        }
    });
    return errors.length === 0;
}
export function validateSchema(schema) {
    if (schema === void 0) { schema = {}; }
    return schema && Object.keys(schema).length > 0;
}
// 校验字段是否为一个合法的变量
export function validateKeyName(v) {
    if (((typeof v === 'string' && v.trim() !== '') || (typeof v !== 'string' && v != null))
        && !v.includes('.')
        && !v.includes('[')
        && !/^([^\x00-\xff]|[a-zA-Z_$])([^\x00-\xff]|[a-zA-Z0-9_$])*$/.test(v)) { // @ts-ignore
        return false;
    }
    return true;
}
