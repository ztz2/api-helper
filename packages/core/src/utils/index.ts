export * from './prettier';
export * from './template';
export * from './util';
export * from './validator';
