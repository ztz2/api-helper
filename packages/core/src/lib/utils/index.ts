import * as prettier from './prettier';
import * as util from './util';
import * as validator from './validator';
import template from './template';

export {
  prettier,
  util,
  validator,
  template,
};
