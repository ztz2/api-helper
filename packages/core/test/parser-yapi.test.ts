import { join } from 'path';

import { ParserYapi } from '../';
import { readJsonFile } from './utils/file';

describe('yapi-parser', () => {
  test('yapi-parser文档解析转换测试', async () => {
    const projectInfo = await readJsonFile(join(__dirname, './resources/yapi/project-info.json'));
    const categoryList = await readJsonFile(join(__dirname, './resources/yapi/category-list.json'));
    const apiList = await readJsonFile(join(__dirname, './resources/yapi/api-list.json'));
    expect(await new ParserYapi({
      projectInfo,
      categoryList,
      apiList,
      autoGenerateId: false,
    }).parser()).toMatchSnapshot('yapi-parser文档解析转换测试');
  });
});
