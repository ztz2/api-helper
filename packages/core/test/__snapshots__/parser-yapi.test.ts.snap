// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`yapi-parser yapi-parser文档解析转换测试: yapi-parser文档解析转换测试 1`] = `
Array [
  Object {
    "basePath": "/v2",
    "categoryList": Array [
      Object {
        "apiList": Array [
          Object {
            "description": "",
            "docURL": "",
            "formDataKeyNameList": Array [],
            "id": "",
            "label": "Finds Pets by status",
            "method": "get",
            "path": "/v2/pet/findByStatus",
            "pathParamKeyNameList": Array [],
            "queryStringKeyNameList": Array [
              "status",
            ],
            "requestDataSchema": ObjectSchema {
              "description": "",
              "enum": Array [],
              "examples": Array [],
              "id": "",
              "keyName": "",
              "label": "",
              "params": Array [
                StringSchema {
                  "description": "Status values that need to be considered for filter",
                  "enum": Array [],
                  "examples": Array [],
                  "id": "",
                  "keyName": "status",
                  "label": "Status values that need to be considered for filter",
                  "params": Array [],
                  "rules": Object {
                    "maxLength": undefined,
                    "minLength": undefined,
                    "pattern": undefined,
                    "required": true,
                  },
                  "title": "",
                  "type": "string",
                },
              ],
              "rules": Object {
                "required": false,
              },
              "title": "",
              "type": "object",
            },
            "requestExtraDataSchema": null,
            "responseDataSchema": ArraySchema {
              "description": "",
              "enum": Array [],
              "examples": Array [],
              "id": "",
              "keyName": "",
              "label": "",
              "params": Array [
                ObjectSchema {
                  "description": "",
                  "enum": Array [],
                  "examples": Array [],
                  "id": "",
                  "keyName": "",
                  "label": "",
                  "params": Array [
                    StringSchema {
                      "description": "",
                      "enum": Array [],
                      "examples": Array [],
                      "id": "",
                      "keyName": "id",
                      "label": "",
                      "params": Array [],
                      "rules": Object {
                        "maxLength": undefined,
                        "minLength": undefined,
                        "pattern": undefined,
                        "required": true,
                      },
                      "title": "",
                      "type": "string",
                    },
                    ObjectSchema {
                      "description": "",
                      "enum": Array [],
                      "examples": Array [],
                      "id": "",
                      "keyName": "category",
                      "label": "",
                      "params": Array [
                        StringSchema {
                          "description": "",
                          "enum": Array [],
                          "examples": Array [],
                          "id": "",
                          "keyName": "id",
                          "label": "",
                          "params": Array [],
                          "rules": Object {
                            "maxLength": undefined,
                            "minLength": undefined,
                            "pattern": undefined,
                            "required": true,
                          },
                          "title": "",
                          "type": "string",
                        },
                        StringSchema {
                          "description": "",
                          "enum": Array [],
                          "examples": Array [],
                          "id": "",
                          "keyName": "name",
                          "label": "",
                          "params": Array [],
                          "rules": Object {
                            "required": true,
                          },
                          "title": "",
                          "type": "string",
                        },
                      ],
                      "rules": Object {
                        "required": true,
                      },
                      "title": "",
                      "type": "object",
                    },
                    StringSchema {
                      "description": "",
                      "enum": Array [],
                      "examples": Array [],
                      "id": "",
                      "keyName": "name",
                      "label": "",
                      "params": Array [],
                      "rules": Object {
                        "required": true,
                      },
                      "title": "",
                      "type": "string",
                    },
                    ArraySchema {
                      "description": "",
                      "enum": Array [],
                      "examples": Array [],
                      "id": "",
                      "keyName": "photoUrls",
                      "label": "",
                      "params": Array [
                        StringSchema {
                          "description": "",
                          "enum": Array [],
                          "examples": Array [],
                          "id": "",
                          "keyName": "",
                          "label": "",
                          "params": Array [],
                          "rules": Object {
                            "required": true,
                          },
                          "title": "",
                          "type": "string",
                        },
                      ],
                      "rules": Object {
                        "maxLength": undefined,
                        "minLength": undefined,
                        "required": true,
                        "uniqueItems": undefined,
                      },
                      "title": "",
                      "type": "array",
                    },
                    ArraySchema {
                      "description": "",
                      "enum": Array [],
                      "examples": Array [],
                      "id": "",
                      "keyName": "tags",
                      "label": "",
                      "params": Array [
                        ObjectSchema {
                          "description": "",
                          "enum": Array [],
                          "examples": Array [],
                          "id": "",
                          "keyName": "",
                          "label": "",
                          "params": Array [
                            StringSchema {
                              "description": "",
                              "enum": Array [],
                              "examples": Array [],
                              "id": "",
                              "keyName": "id",
                              "label": "",
                              "params": Array [],
                              "rules": Object {
                                "maxLength": undefined,
                                "minLength": undefined,
                                "pattern": undefined,
                                "required": true,
                              },
                              "title": "",
                              "type": "string",
                            },
                            StringSchema {
                              "description": "",
                              "enum": Array [],
                              "examples": Array [],
                              "id": "",
                              "keyName": "name",
                              "label": "",
                              "params": Array [],
                              "rules": Object {
                                "required": true,
                              },
                              "title": "",
                              "type": "string",
                            },
                          ],
                          "rules": Object {
                            "required": true,
                          },
                          "title": "",
                          "type": "object",
                        },
                      ],
                      "rules": Object {
                        "maxLength": undefined,
                        "minLength": undefined,
                        "required": true,
                        "uniqueItems": undefined,
                      },
                      "title": "",
                      "type": "array",
                    },
                    StringSchema {
                      "description": "pet status in the store",
                      "enum": Array [
                        "available",
                        "pending",
                        "sold",
                      ],
                      "examples": Array [],
                      "id": "",
                      "keyName": "status",
                      "label": "pet status in the store",
                      "params": Array [],
                      "rules": Object {
                        "required": true,
                      },
                      "title": "",
                      "type": "string",
                    },
                  ],
                  "rules": Object {
                    "required": true,
                  },
                  "title": "",
                  "type": "object",
                },
              ],
              "rules": Object {
                "maxLength": undefined,
                "minLength": undefined,
                "required": true,
                "uniqueItems": undefined,
              },
              "title": "",
              "type": "array",
            },
            "title": "Finds Pets by status",
          },
          Object {
            "description": "",
            "docURL": "",
            "formDataKeyNameList": Array [
              "additionalMetadata",
              "file",
            ],
            "id": "",
            "label": "uploads an image",
            "method": "post",
            "path": "/v2/pet/{petId}/uploadImage",
            "pathParamKeyNameList": Array [
              "petId",
            ],
            "queryStringKeyNameList": Array [],
            "requestDataSchema": ObjectSchema {
              "description": "",
              "enum": Array [],
              "examples": Array [],
              "id": "",
              "keyName": "",
              "label": "",
              "params": Array [
                StringSchema {
                  "description": "ID of pet to update",
                  "enum": Array [],
                  "examples": Array [],
                  "id": "",
                  "keyName": "petId",
                  "label": "ID of pet to update",
                  "params": Array [],
                  "rules": Object {
                    "maxLength": undefined,
                    "minLength": undefined,
                    "pattern": undefined,
                    "required": true,
                  },
                  "title": "",
                  "type": "string",
                },
                StringSchema {
                  "description": "Additional data to pass to server",
                  "enum": Array [],
                  "examples": Array [],
                  "id": "",
                  "keyName": "additionalMetadata",
                  "label": "Additional data to pass to server",
                  "params": Array [],
                  "rules": Object {
                    "maxLength": undefined,
                    "minLength": undefined,
                    "pattern": undefined,
                    "required": false,
                  },
                  "title": "",
                  "type": "string",
                },
                StringSchema {
                  "description": "file to upload",
                  "enum": Array [],
                  "examples": Array [],
                  "id": "",
                  "keyName": "file",
                  "label": "file to upload",
                  "params": Array [],
                  "rules": Object {
                    "maxLength": undefined,
                    "minLength": undefined,
                    "pattern": undefined,
                    "required": false,
                  },
                  "title": "",
                  "type": "File",
                },
              ],
              "rules": Object {
                "required": false,
              },
              "title": "",
              "type": "object",
            },
            "requestExtraDataSchema": null,
            "responseDataSchema": ObjectSchema {
              "description": "",
              "enum": Array [],
              "examples": Array [],
              "id": "",
              "keyName": "",
              "label": "",
              "params": Array [
                NumberSchema {
                  "description": "",
                  "enum": Array [],
                  "examples": Array [],
                  "id": "",
                  "keyName": "code",
                  "label": "",
                  "params": Array [],
                  "rules": Object {
                    "exclusiveMaximum": undefined,
                    "exclusiveMinimum": undefined,
                    "maximum": undefined,
                    "minimum": undefined,
                    "multipleOf": undefined,
                    "required": true,
                  },
                  "title": "",
                  "type": "number",
                },
                StringSchema {
                  "description": "",
                  "enum": Array [],
                  "examples": Array [],
                  "id": "",
                  "keyName": "type",
                  "label": "",
                  "params": Array [],
                  "rules": Object {
                    "required": true,
                  },
                  "title": "",
                  "type": "string",
                },
                StringSchema {
                  "description": "",
                  "enum": Array [],
                  "examples": Array [],
                  "id": "",
                  "keyName": "message",
                  "label": "",
                  "params": Array [],
                  "rules": Object {
                    "required": true,
                  },
                  "title": "",
                  "type": "string",
                },
              ],
              "rules": Object {
                "required": true,
              },
              "title": "",
              "type": "object",
            },
            "title": "uploads an image",
          },
          Object {
            "description": "",
            "docURL": "",
            "formDataKeyNameList": Array [],
            "id": "",
            "label": "Add a new pet to the store",
            "method": "post",
            "path": "/v2/pet",
            "pathParamKeyNameList": Array [],
            "queryStringKeyNameList": Array [],
            "requestDataSchema": ObjectSchema {
              "description": "",
              "enum": Array [],
              "examples": Array [],
              "id": "",
              "keyName": "",
              "label": "",
              "params": Array [
                StringSchema {
                  "description": "",
                  "enum": Array [],
                  "examples": Array [],
                  "id": "",
                  "keyName": "id",
                  "label": "",
                  "params": Array [],
                  "rules": Object {
                    "maxLength": undefined,
                    "minLength": undefined,
                    "pattern": undefined,
                    "required": false,
                  },
                  "title": "",
                  "type": "string",
                },
                ObjectSchema {
                  "description": "",
                  "enum": Array [],
                  "examples": Array [],
                  "id": "",
                  "keyName": "category",
                  "label": "",
                  "params": Array [
                    StringSchema {
                      "description": "",
                      "enum": Array [],
                      "examples": Array [],
                      "id": "",
                      "keyName": "id",
                      "label": "",
                      "params": Array [],
                      "rules": Object {
                        "maxLength": undefined,
                        "minLength": undefined,
                        "pattern": undefined,
                        "required": false,
                      },
                      "title": "",
                      "type": "string",
                    },
                    StringSchema {
                      "description": "",
                      "enum": Array [],
                      "examples": Array [],
                      "id": "",
                      "keyName": "name",
                      "label": "",
                      "params": Array [],
                      "rules": Object {
                        "required": false,
                      },
                      "title": "",
                      "type": "string",
                    },
                  ],
                  "rules": Object {
                    "required": false,
                  },
                  "title": "",
                  "type": "object",
                },
                StringSchema {
                  "description": "",
                  "enum": Array [],
                  "examples": Array [],
                  "id": "",
                  "keyName": "name",
                  "label": "",
                  "params": Array [],
                  "rules": Object {
                    "required": true,
                  },
                  "title": "",
                  "type": "string",
                },
                ArraySchema {
                  "description": "",
                  "enum": Array [],
                  "examples": Array [],
                  "id": "",
                  "keyName": "photoUrls",
                  "label": "",
                  "params": Array [
                    StringSchema {
                      "description": "",
                      "enum": Array [],
                      "examples": Array [],
                      "id": "",
                      "keyName": "",
                      "label": "",
                      "params": Array [],
                      "rules": Object {
                        "required": false,
                      },
                      "title": "",
                      "type": "string",
                    },
                  ],
                  "rules": Object {
                    "maxLength": undefined,
                    "minLength": undefined,
                    "required": true,
                    "uniqueItems": undefined,
                  },
                  "title": "",
                  "type": "array",
                },
                ArraySchema {
                  "description": "",
                  "enum": Array [],
                  "examples": Array [],
                  "id": "",
                  "keyName": "tags",
                  "label": "",
                  "params": Array [
                    ObjectSchema {
                      "description": "",
                      "enum": Array [],
                      "examples": Array [],
                      "id": "",
                      "keyName": "",
                      "label": "",
                      "params": Array [
                        StringSchema {
                          "description": "",
                          "enum": Array [],
                          "examples": Array [],
                          "id": "",
                          "keyName": "id",
                          "label": "",
                          "params": Array [],
                          "rules": Object {
                            "maxLength": undefined,
                            "minLength": undefined,
                            "pattern": undefined,
                            "required": false,
                          },
                          "title": "",
                          "type": "string",
                        },
                        StringSchema {
                          "description": "",
                          "enum": Array [],
                          "examples": Array [],
                          "id": "",
                          "keyName": "name",
                          "label": "",
                          "params": Array [],
                          "rules": Object {
                            "required": false,
                          },
                          "title": "",
                          "type": "string",
                        },
                      ],
                      "rules": Object {
                        "required": false,
                      },
                      "title": "",
                      "type": "object",
                    },
                  ],
                  "rules": Object {
                    "maxLength": undefined,
                    "minLength": undefined,
                    "required": false,
                    "uniqueItems": undefined,
                  },
                  "title": "",
                  "type": "array",
                },
                StringSchema {
                  "description": "pet status in the store",
                  "enum": Array [
                    "available",
                    "pending",
                    "sold",
                  ],
                  "examples": Array [],
                  "id": "",
                  "keyName": "status",
                  "label": "pet status in the store",
                  "params": Array [],
                  "rules": Object {
                    "required": false,
                  },
                  "title": "",
                  "type": "string",
                },
              ],
              "rules": Object {
                "required": false,
              },
              "title": "",
              "type": "object",
            },
            "requestExtraDataSchema": null,
            "responseDataSchema": null,
            "title": "Add a new pet to the store",
          },
        ],
        "description": "Everything about your Pets",
        "id": "",
        "name": "pet",
      },
      Object {
        "apiList": Array [],
        "description": "未设置分组的接口集合",
        "id": "",
        "name": "未知分组",
      },
    ],
    "description": "",
    "documentServerUrl": "",
    "documentVersion": "2.0",
    "id": "",
    "title": "智慧商城",
    "version": "last",
  },
]
`;
